pkg/controller/namespace/namespace_controller.go

```diff
@@ -208,6 +193,35 @@ func (r *ReconcileNamespace) Reconcile(request reconcile.Request) (reconcile.Res
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileNamespace) isControlledByWorkspace(namespace *corev1.Namespace) (bool, error) {
```

Comment:
function name and what this function does doesn't match, `isControllerByWorkspace` should just test on a namespace to see if it belongs a workspace, but now it also does cleanup work.